stage.go

```diff
@@ -1,15 +1,40 @@
 package progress
 
+import "sync/atomic"
+
 type Stager interface {
 	Stage() string
 }
 
 type Stage struct {
```

Comment:
I think adding a new type for an `AsyncStage` or something similar instead of breaking existing consumers -- that would lean nicely into the consumption-by-interface approach we have today.